docs/data/base/getting-started/customization/PrepareForSlot.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { prepareForSlot } from '@mui/base/utils';
+import { Button } from '@mui/base/Button';
+import Link from 'next/link';
+
+const LinkSlot = prepareForSlot(Link);
+
+export default function PrepareForSlot() {
+  return (
+    <Button href={'/'} slots={{ root: LinkSlot }}>
+      Home
+    </Button>
+  );
+}
```

Comment:
I doubt this exampls works in real life. For example, this returns an exception:

```jsx
<Button href="/" slots={{ root: LinkSlot }} prefetch={false}>
```

https://nextjs.org/docs/pages/api-reference/components/link#prefetch

Maybe we should remove the `prepareForSlot()` API and instead document this?

```jsx
import * as React from 'react';
import { Button, ButtonProps } from '@mui/base/Button';
import Link, { LinkProps } from 'next/link';

const ButtonLink = React.forwardRef<HTMLAnchorElement, ButtonProps & LinkProps>(
  function ButtonLink(props, ref) {
    const {
      // @ts-expect-error
      ownerState,
      ...other
    } = props;
    return <Button ref={ref as any} slots={{ root: Link }} {...other} />;
  },
);

export default function PrepareForSlot() {
  return (
    <ButtonLink href="/" prefetch={false}>
      Home
    </ButtonLink>
  );
}
```

Reply:
We dropped the implicit polymorphism, but it's possible to make it explicit, as described in https://mui.com/base-ui/react-button/#custom-structure

The example should use it:
```tsx
<Button<typeof LinkSlot> href="/" slots={{ root: LinkSlot }} prefetch={false}>
```

Reply:
Oh, I didn't know this. This sounds quite interesting.
What happens once a component is customized with `styled()`, does it still work?

Could we apply this to our example? As I think the initial point I raise stays true. 

Actually, once applied repeatedly, it feels like this asks for a wrapper component to keep it more nominal. Once you introduce a wrapper, you are better not to use `prepareForSlot` as it creates fewer components, better performance.